src/utils/cloneWithProps/cloneWithProps.ts

```diff
@@ -0,0 +1,21 @@
+// REACT
+import * as React from 'react';
+// VENDOR
+import classNames from 'classnames';
+
+interface Props {
+    className?: string;
+    [key: string]: any;
+}
+
+export const cloneWithProps = (
+    element: React.ReactElement<any>,
+    props: Props = {}
+) => {
+    const { className, ...rest } = props;
+    return React.cloneElement(element, {
+        ...rest,
+        ...element.props,
+        className: classNames(className, element.props.className),
+    });
+};
```

Comment:
I think a comment block explaining why you built such a robust cloning service would be helpful.